mailpoet/lib/Automation/Engine/Control/SubjectTransformerHandler.php

```diff
@@ -0,0 +1,142 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Control;
+
+use MailPoet\Automation\Engine\Data\Automation;
+use MailPoet\Automation\Engine\Data\AutomationRun;
+use MailPoet\Automation\Engine\Data\Step as StepData;
+use MailPoet\Automation\Engine\Data\Subject as SubjectData;
+use MailPoet\Automation\Engine\Integration\Step;
+use MailPoet\Automation\Engine\Integration\SubjectTransformer;
+use MailPoet\Automation\Engine\Registry;
+use MailPoet\Automation\Engine\Storage\AutomationStorage;
+
+class SubjectTransformerHandler {
+
+  /* @var Registry */
+  private $registry;
+
+  /* @var AutomationStorage */
+  private $automationStorage;
+
+  public function __construct(
+    Registry $registry,
+    AutomationStorage $automationStorage
+  ) {
+    $this->registry = $registry;
+    $this->automationStorage = $automationStorage;
+  }
+
+  /** @return string[] */
+  public function subjectKeysForAutomation(Automation $automation): array {
+    $configuredTriggers = array_filter(
+      $automation->getSteps(),
+      function (StepData $step): bool {
+        return $step->getType() === StepData::TYPE_TRIGGER;
+      }
+    );
+
+    $triggers = array_filter(array_map(
+      function(StepData $step): ?Step {
+        return $this->registry->getStep($step->getKey());
```

Comment:
Should we filter out the triggers that are not in the registry or throw an error here? Maybe this is OK because on activation the validator would take care of this?